client/directclient.go

```diff
@@ -63,15 +65,35 @@ func (c *ProxyDirectClient) quorumResponse(reqs ...*http.Request) (http.Header,
 	}
 	quorum := int(math.Ceil(float64(len(reqs)) / 2.0))
 	responseClasses := []int{0, 0, 0, 0, 0, 0}
+	responseCount := 0
+	var chosenResponse *QuorumResponseEntry
 	for response := range responses {
+		responseCount++
 		class := response.StatusCode / 100
 		if class <= 5 {
 			responseClasses[class]++
 			if responseClasses[class] >= quorum {
-				return response.Headers, response.StatusCode
+				chosenResponse = response
+				break
+			}
+		}
+	}
+	// give any pending requests *some* chance to finish
+	if responseCount < len(reqs) {
```

Comment:
Should this be a while loop, so it waits for all responses?  I think I read this as a while the first time I looked through it, and that's why I thought it would wait up to 200 ms per outstanding request.